### DISCUSSION TITLE
Rotating Terminal Background Functionality

### DISCUSSION THREAD

**Discussion by iton0** [wezterm/wezterm#4345] [Ideas]
Hey everyone,

I've been using wezterm as my terminal emulator for a while now, and I've been exploring its customization options. One feature that caught my attention recently is the ability to set a background image for the terminal. However, I was wondering if there's a way to make the background image change or rotate at intervals.

I think having a rotating terminal background could add a nice touch to my workflow and keep things visually interesting. Plus, it might be a fun project to work on. Has anyone tried implementing rotating background functionality in wezterm, or do you have any ideas on how this could be achieved? I'd love to hear your thoughts and experiences with this.

**Comment by wez** [comment]
https://wezfurlong.org/wezterm/config/lua/wezterm/get_builtin_color_schemes.html has an example of picking a random color scheme. You could use a similar technique to set the background instead.

You could also use https://wezfurlong.org/wezterm/config/lua/wezterm.time/call_after.html to periodically reload the config and select a different background.
